invenio_previewer/extensions/tar.py

```diff
@@ -0,0 +1,89 @@
+from __future__ import absolute_import, print_function
+
+import os
+import tarfile
+
+import cchardet as chardet
+from flask import current_app, render_template
+from six import binary_type
+
+from .._compat import text_type
+from ..proxies import current_previewer
+
+previewable_extensions = ['tar', 'gz', 'bz2', 'tgz', 'tbz', 'tbz2']
+
+
+def make_tree(file):
+    """Create tree structure from tar archives."""
+    max_files_count = current_app.config.get('PREVIEWER_ZIP_MAX_FILES', 1000)
+    tree = {'type': 'folder', 'id': -1, 'children': {}}
+
+    try:
+        with file.open() as fp, tarfile.open(fileobj=fp) as tf:
+            # Detect filenames encoding.
+            sample = ' '.join(tf.getnames()[:max_files_count])
+            if not isinstance(sample, binary_type):
+                sample = sample.encode('utf-16be')
+            encoding = chardet.detect(sample).get('encoding', 'utf-8')
+            for i, info in enumerate(tf.getmembers()):
+                if i > max_files_count:
+                    raise BufferError('Too many files inside the TAR file.')
```

Comment:
Looks like a fine implementation to me. Leaving a review on the code based on what I know of tarfile standard library and Python

The code invokes only `TarFile.getnames()` and `TarfFile.getmembers()` method. These look at the list of the files included (equivalent to `tar tvf` posix / unix command) and does not extract the files. Hope that answers @lnielsen's question.

https://docs.python.org/3.10/library/tarfile.html?highlight=tarfile#tarfile.TarFile.getmembers